logstage/logstage-monix/src/main/scala/logstage/LogstageCats.scala

```diff
@@ -0,0 +1,30 @@
+package logstage
+
+import cats.Monad
+import com.github.pshirshov.izumi.functional.mono.SyncSafe
+import com.github.pshirshov.izumi.logstage.api.Log.{CustomContext, Entry, Message}
+import cats.implicits._
+import com.github.pshirshov.izumi.fundamentals.reflection.CodePositionMaterializer
+import logstage.LogCreateIO.LogCreateIOSyncSafeInstance
+
+object LogstageCats {
+
+  def withDynamicContext[F[_]: Monad: SyncSafe](logger: IzLogger, dynamic: F[CustomContext]): LogIO[F] = {
+
+    def withLogger[T](f: IzLogger => T) : F[T] = {
+      dynamic.flatMap(ctx => SyncSafe[F].syncSafe(logger.withCustomContext(ctx)).map(f))
```

Comment:
```suggestion
      dynamic.flatMap(ctx => SyncSafe[F].syncSafe(f(logger.withCustomContext(ctx))))
```
`logger.withContext` is not the operation that needs wrapping, `f` is